### PR TITLE
Books detail

### PR DESCRIPTION
**Pull request by eledwinn**
## Summary

1. Add ui-router
2. Add book detail page
3. Refactor book box component (book-item)

## Screenshot
image

## Trello Card
https://trello.com/c/waEXCySZ/17-book-detail

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,11 @@
...
 ## System ##
 .DS_Store
+
+logs/*
+!.gitkeep
+app/lib/
+node_modules/
+tmp
+.DS_Store
+.idea
```
Comment (FrankIglesias) [inline .gitignore]: add EOL

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+// Declare app level module which depends on views, and core components
```
Comment (FrankIglesias) [inline app/app.js]: delete this comment

File: app/app.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+angular.module('wBooksApp', [
+  'ui.router',
+  'core.book',
+  'bookItem',
+  'booksList',
+  'bookDetail',
+  'wInput'
+]).
+config(['$stateProvider', '$urlRouterProvider', function($stateProvider, $urlRouterProvider) {
+  // An array of state definitions
```
Comment (FrankIglesias) [inline app/app.js]: same

File: app/book-detail/book-detail.template.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      <div class="book-detail-content mar-l-3">
+        <h1 class="book-detail-title">{{$ctrl.book.title}}</h1>
+        <p class="book-detail-author">{{$ctrl.book.author}}</p>
+        <p class="book-detail-year">{{$ctrl.book.year}}</p>
+        <p class="book-detail-genre">{{$ctrl.book.genre}}</p>
+        <p class="book-detail-resume">Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.</p>
+        <button class="btn btn-primary book-detail-button">Alquilar</button>
+      </div>
+  </div>
+  <hr class="main-body splitter">
```
Comment (FrankIglesias) [inline app/book-detail/book-detail.template.html]: use a border for this

File: app/book-detail/book-detail.template.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  <div class="main-body">
+    <h2 class="subtitle">Sugerencias</h2>
+    <div class="book-items-inline">
+      <book-item book="{ id: book }" small="true" ng-repeat="book in [1,2,3,4]"></book-item>
+    </div>
+  </div>
+  <hr class="main-body splitter">
+  <div class="main-body">
```
Comment (FrankIglesias) [inline app/book-detail/book-detail.template.html]: move the margin o whatever style you are applying to the subtitle

File: app/book-detail/book-detail.template.html [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      </div>
+  </div>
+  <hr class="main-body splitter">
+  <div class="main-body">
+    <h2 class="subtitle">Sugerencias</h2>
+    <div class="book-items-inline">
+      <book-item book="{ id: book }" small="true" ng-repeat="book in [1,2,3,4]"></book-item>
+    </div>
+  </div>
+  <hr class="main-body splitter">
```
Comment (FrankIglesias) [inline app/book-detail/book-detail.template.html]: use a border for this

File: app/common/w-input/w-input.template.html [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    type="text"
+    class="w-input"
+    placeholder="{{$ctrl.placeholder}}"
+    ng-model="$ctrl.inputValue"
+    ng-keypress="$ctrl.onKeyPress($event)"
+    ng-change="$ctrl.inputValueChanged()">
+  <button ng-if="!$ctrl.items"
+    class="w-input-button"
+    ng-click="$ctrl.onSearch()">
+    <span class="fa fa-search"></span>
```
Comment (FrankIglesias) [inline app/common/w-input/w-input.template.html]: This is for the icon isn't it? Add the class to the button instead

File: app/common/w-input/w-input.template.html [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    placeholder="{{$ctrl.placeholder}}"
+    ng-model="$ctrl.inputValue"
+    ng-keypress="$ctrl.onKeyPress($event)"
+    ng-change="$ctrl.inputValueChanged()">
+  <button ng-if="!$ctrl.items"
+    class="w-input-button"
+    ng-click="$ctrl.onSearch()">
+    <span class="fa fa-search"></span>        
+  </button>
+
```
Comment (FrankIglesias) [inline app/common/w-input/w-input.template.html]: delete extra line

File: app/index.html [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+<head>
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <title>WBooks</title>
+  <meta name="description" content="">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="application.css">
+</head>
+<body>
+  <!-- <ul class="menu">
```
Comment (FrankIglesias) [inline app/index.html]: remove some extra comments

File: app/scss/book-detail.scss [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+
+.book-detail-author {
+  margin-bottom: 10px;
+}
+
+.book-detail-year {
+  margin-bottom: 5px;
+}
+
+.book-detail-author, .book-detail-year {
```
Comment (FrankIglesias) [inline app/scss/book-detail.scss]: .book-detail-author,
book-detail-year {
}

File: app/scss/book-detail.scss [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    font-size: 1.1rem;
```
Comment (FrankIglesias) [inline app/scss/book-detail.scss]: use px

File: app/scss/book-detail.scss [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+.book-detail-genre {
+  margin-bottom: 10px;
+}
+
+.book-detail-button {
+  font-size: 1.2rem;
```
Comment (FrankIglesias) [inline app/scss/book-detail.scss]: use px

File: app/scss/book-detail.scss [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+.book-detail-content {
+  align-items: flex-start;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+}
+
+.book-detail-title {
+  font-size: 1.5rem;
```
Comment (FrankIglesias) [inline app/scss/book-detail.scss]: use px

File: app/scss/book-item.scss [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+.book-items-inline {
+  justify-content: flex-start;
+}
+
+.book-item {
+  display: flex;
+  flex-direction: column;
+  width: 128px;
+  margin: 0 30px 30px 30px;
```
Comment (FrankIglesias) [inline app/scss/book-item.scss]: margin: 0 30px 30px;

File: app/scss/equalizer.scss [hunk 15]
```diff
@@ -0,0 +1,92 @@
+$equalizer-black: #000;
```
Comment (FrankIglesias) [inline app/scss/equalizer.scss]: use [wolox equalizer ](https://www.npmjs.com/package/wolox-equalizer)

File: app/app.js [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    { 
+      name: 'bookDetail', 
+      // This state takes a URL parameter called personId
+      url: '/books/{id}', 
+      component: 'bookDetail',
+      // This state defines a 'person' resolve
+      // It delegates to the PeopleService, passing the personId parameter
+      resolve: {
+        book: function(Books, $stateParams) {
+
```
Comment (FrankIglesias) [inline app/app.js]: delete extra line

File: app/app.js [hunk 17]
```diff
@@ -0,0 +1,46 @@
+'use strict';
```
Comment (FrankIglesias) [inline app/app.js:1]: Delete comments of this file

File: app/scss/styles.scss [hunk 18]
```diff
@@ -0,0 +1,65 @@
+* {
```
Comment (FrankIglesias) [inline app/scss/styles.scss]: maybe you will have to divide this classes into separate files according to it's usage

File: app/scss/w-input.scss [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+.w-input-button {
+  background: $green;
+  bottom: 0;
+  color: $white;
+  padding: 0 10px;
+  position: absolute;
+  right: 0;
+  top: 0;
+
+  .fa {
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: why this is fa?

File: app/scss/w-input.scss [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+
+.w-select {
+  display: inline-block;
+  padding-right: 3.5em;
+  background-image: linear-gradient(50deg, transparent 50%, gray 50%), linear-gradient(125deg, gray 50%, transparent 50%);
+  background-position: calc(100% - 22px) calc(1em + 2px), calc(100% - 12px) calc(1em + 2px), calc(100% - 2.5em) 0.5em;
+  background-size: 7px 10px, 10px 10px, 1px 2em;
+  background-repeat: no-repeat;
+
+  /* reset */
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: this should be equalizer's job

File: app/scss/w-input.scss [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+  -webkit-box-sizing: border-box;
+  -moz-box-sizing: border-box;
+  box-sizing: border-box;
+  -webkit-appearance: none;
+  -moz-appearance: none;
+}
+
+.w-select:-moz-focusring {
+  color: transparent;
+  text-shadow: 0 0 0 #000;
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: use color variable

File: package.json [hunk 22]
```diff
@@ -0,0 +1,50 @@
+{
+  "name": "wbooks",
+  "version": "1.0.0",
+  "description": "Wolox Books Training App",
+  "main": "index.js",
+  "scripts": {
+    "postinstall": "npm run copy-libs",
```
Comment (FrankIglesias) [inline package.json:7]: delete some extra scripts

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: app/index.html (hunk 9 above)
Comment (eledwinn) [reply app/index.html]: This is ready in "books-pug" branch

File: package.json (hunk 22 above)
Comment (eledwinn) [reply package.json:7]: This is ready in "books-gulp" branch

File: app/book-item/book-item.component.js [hunk 23]
```diff
@@ -0,0 +1,15 @@
...
+  module('bookItem', []).
+  component('bookItem', {
+    templateUrl: 'book-item/book-item.template.html',
+    bindings: { 
+      book: '<',
+      small: '@'
+    },
+    controller: function BooksListController() {
+      var ctrl = this;
+
```
Comment (FrankIglesias) [inline app/book-item/book-item.component.js:13]: delete this line

File: app/scss/book-item.scss [hunk 24]
```diff
@@ -1,26 +1,33 @@
-.book-items {
+.book-items, .book-items-inline {
```
Comment (FrankIglesias) [inline app/scss/book-item.scss:1]: .book-items,
.book-items-inline

File: app/scss/subtitle.scss [hunk 25]
```diff
@@ -0,0 +1,5 @@
+.subtitle {
+  color: $green;
+  font-size: 1.3rem;
```
Comment (FrankIglesias) [inline app/scss/subtitle.scss:3]: use px

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

File: app/book-item/book-item.component.js (hunk 23 above)
Comment (eledwinn) [reply app/book-item/book-item.component.js:13]: This is ready in books-bootstrap branch

File: app/scss/book-item.scss (hunk 24 above)
Comment (eledwinn) [reply app/scss/book-item.scss:1]: This is ready in books-bootstrap branch

File: app/scss/subtitle.scss (hunk 25 above)
Comment (eledwinn) [reply app/scss/subtitle.scss:3]: This is ready in books-bootstrap branch

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**